### ISSUE TITLE
Diol

### ISSUE THREAD

**Issue by OlzhabaevSh** [dotnet-foundation/projects#294]
## 1.   General Information

**Project Name:** 
Diol

**License:** 
MIT

**Contributor (Company, Organization or individual name(s)):**
Shynggyzkhan Olzhabayev 
Anton Tishin
Sergey Antonov

**Existing OSS Project? (Yes/No):** 
No

**Source Code URL:** 
https://github.com/Dotnet-IO-logger/core

**Project Homepage URL (if different):**

**Project Transfer Signatories:**
Full legal name and __email address__ required of the individual(s) with the authority to transfer or contribute the project to the .NET Foundation. Note that if you'd prefer not to include this in the public application, it can be submitted via e-mail to contact@dotnetfoundation.org (referencing this issue number).

Shynggyzkhan Olzhabayev
olzhabaev.sh@outlook.com

## 2.   Description
**Please provide a brief statement about your project in terms that are understandable to the target consumer of the library or project, i.e. an elevator pitch for the project:**

Diol is a Visual Studio extension that allows you to view your http logs in realtime. With Diol, you can monitor and debug your dotnet applications more easily and efficiently. Diol is free to use and the project is open source, so you can contribute to its development and improvement. Diol aims to provide you with a fun and convenient way to work with http logs in Visual Studio.

**Please provide a 1 sentence (<140 character) summary of your project to help users when searching the .NET Foundation projects**

New Visual Studio extension for monitoring dotnet application's http logs in realtime.

## 3.   Project Governance
Please complete this section about who will be maintaining the open source project and how it will run.
**Project Lead:**
(Who is the primary contact point for the community and the .NET Foundation when discussing governance of the project.)

Name:   Shynggyzkhan Olzhabayev
 Email:   olzhabaev.sh@outlook.com
 GitHub Profile URL: https://github.com/OlzhabaevSh

**Committers:**

Which individuals have commit / write access to the repository, what is their GitHub ID and who is their employer (if contributions are on behalf of an employer)

Shynggyzkhan Olzhabayev. OlzhabaevSh
Anton Tishin. atishin
Sergey Antonov. Scogun

**Governance Model:**

Please describe how new code changes are proposed to the project, how those changes are reviewed and how a decision is made to accept proposed changes. Also describe the process for identifying and appointing new committers.

For this moment we are going to use issue tracker and pr for adding new changes. 
The process is new and we have no vision about future so we are open to be flexible.

**CLA**

If already an OSS project, was a Contribution License Agreement in place for contributions accepted? How does the project check who has signed one?

**CLA Notification Alias**
Provide an email address that will receive CLA related notifications from the .NET Foundation CLA automation

olzhabaev.sh@outlook.com

**Assignment Model.**  Under the .NET Foundation assignment model, project ownership and other intellectual property is assigned to the .NET Foundation and the .NET Foundation agrees to grantback a license to the contributor(s). 

**Contribution Model.**  Under the .NET Foundation contribution model, a project retains ownership of the copyright, but grants the .NET Foundation a broad license to the project’s code and other intellectual property. The project also confirms that the project’s submissions to .NET Foundation are its own original work (there are also instructions for any third party materials that might be included).

We would prefer **Contribution Model.**

## 4.   Repository Layout
The .NET Foundation host guidance for new projects and details on recommended structure here:
[https://github.com/dotnet/home/tree/master/guidance](https://github.com/dotnet/home/tree/master/guidance)

Note that the open source repository should be the master where changes are made by the core development team using the same PR process that is used for non-committer contributions.

Please define below any changes you would want to make to your repositories as part of the process of joining the .NET Foundation

## 5.  Eligibility Criteria
Please complete the following for your project
* The project is built on the .NET platform and/or creates value within the .NET ecosystem.
- [ x] Yes 
- [ ] No
* The project produces source code for distribution to the public at no charge.
- [x ] Yes 
- [ ] No
* The project's code is easily discoverable and publicly accessible (preferably on GitHub).
- [x ] Yes 
- [ ] No
* The project contains a build script that can produce deployable artifacts that are identical to the official deployable artifacts, with the exception of code signing (Exception may be granted for strong name keys, though strongly encouraged to be committed. Exception relies on OSS signing being in the build script for public builds).
- [x ] Yes 
- [ ] No
* When applicable, project must use reproducible build settings in its toolchain.
- [ ] Yes 
- [ x] No
* The project uses [Source Link.](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink)
- [ ] Yes 
- [ x] No
* The project uses either embedded PDBs or publish symbol packages to NuGet (if applicable).
- [ ] Yes 
- [x ] No
* The project code signs their artifacts as appropriate.
- [ ] Yes 
- [x ] No
* The project organization has 2FA enabled. Requiring 2FA must be done as part of onboarding if not already enabled.
- [x ] Yes
- [ ] No
* Libraries that are mandatory dependencies of the project are offered under a standard, permissive open source license which has been approved by the .NET Foundation (exceptions include a dependency that is required by the target platform where no alternative open source dependency is available such as the .NET Framework or a hardware specific library).
- [x ] Yes 
- [ ] No
* Committers are bound by a Contributor License Agreement (CLA) and/or are willing to embrace the .NET Foundation's CLA when the project becomes a Member.
- [ x] Yes 
- [ ] No
* The copyright ownership of everything that the project produces is clearly defined and documented.
- [x ] Yes 
- [ ] No
* The project has a public issue tracker where the status of any defect can be easily obtained.
- [x ] Yes 
- [ ] No
* The project has a published Security Policy.
- [x ] Yes 
- [ ] No
* The project has a home page which provides high level information about its status and purpose.
- [x ] Yes 
- [ ] No
* The project has a public communication channel where community members can engage with maintainers.
- [x ] Yes 
- [ ] No
* The project has a publicly available location where members can review and contribute to documentation.
- [x ] Yes 
- [ ] No

## 6.   PR Plan
Please summarize the public relations plan for the announcement when joining the foundation (and releasing as open source if appropriate). What is the main story we wish to promote, through what channels, what issues should we be aware of?  For significant news events then please also work with your .NET Foundation contact to ensure a [full PR plan](https://dotnetfoundation.sharepoint.com/Shared%20Documents/PR/Communications%20Plan%20TEMPLATE.docx?web=1) is developed.
The project is new and we have no plan. 
Everything is based on libs under MIT and we have not released yet.

## 7.   Infrastructure Requirements
Please describe any infrastructure requirements for the project. For example, how will build servers be operated? Any web hosting or service hosting requirements? Do we need to set up SSL certificates or provide Authenticode Code Signing arrangement for releases?
For VSIX distribution we are planing to use VS market as a platform so we don't really need to have our own servers. 
Everything works on client's machine. 
Only one thing we really need:
Code sign certificate. 
Because our extension depends on additional service (we call it as DiolBackendService which anyway lives in client machine as well), users will have to isntall that service as well. Without Code Sign certificate it will show potential issues

## 8.   Additional Notes
Please provide any additional information required or use this area for notes during the onboarding process. If this open source project has similarities with any other projects in this space then please detail them and why this project is different. If there are any potential issues that you feel the project might need help with early on then also state them here and discuss with your .NET Foundation Contact.
Diol actually does the same what frontend engineers already have (browser -> dev tools -> monitoring). 
We tried to bring the same expirience to the .net world!
We also know that there is Diagnostic Tool in VS which can do a bit the same, but we beleive that Diol can provide better and more focus expirience.

**Comment by OlzhabaevSh** [comment]
Hey there,
Let me know please if anything else is required from us to continue the process?

**Comment by sbwalker** [comment]
The Project Committee will review the application at its next scheduled meeting on July 13

**Comment by OlzhabaevSh** [comment]
Thank you for reply!

**Comment by sbwalker** [comment]
The Project Committee recently reviewed your application. The Project Committee Charter (https://github.com/dotnet-foundation/projects) contains activity criteria which is used to evaluate project applications. This project does not currently have activity or adoption metrics at a level which would allow for an application approval - however based on the broad applicability, it will be included in the Seed category (https://dnfprojects.org/!/Jun-13-2023/Jul-13-2023/seed/pr) for consideration as a Member project in the future.

**Comment by OlzhabaevSh** [comment]
Hello @sbwalker 
Thank you for your update. 
Just to clarify, can we get some support regarding improvements for our repo to be able to achive that criterias? 
And what should be our next steps?
Thank you

**Comment by sbwalker** [comment]
@OlzhabaevSh in order to graduate to  Member project status, a Seed project needs to demonstrate that it has achieved sufficient traction in the .NET ecosystem (both in terms of users as well as contributors), that the project activity (ie. Commits, PRs, Issues resolved, etc...) have been sustained for a sufficient period of time to demonstrate ongoing commitment, and that the project has cultivated an active user/developer community around it. These are standard activities which maintainer(s) of open source projects must provide in order to establish their credibility and momentum. Hopefully this provides the clarification you are looking for.

**Comment by OlzhabaevSh** [comment]
Hey @sbwalker 
Thank you for clear explanation. 
Just to clarify, we can not initiate getting code sign certificate until the onboarding finish, is it correct?
One part of our solution requires to be installed on client machine (Backend service). 
Of course, users still are able to build it from source code and install manually, but the process is complex a bit.

**Comment by sbwalker** [comment]
@OlzhabaevSh You are correct that services such as code signing are reserved for Member projects. I can inquire internally to determine if it may be possible to get an exception for your project.
